weblab_cg/_generator_chaste_cg.py

```diff
@@ -601,131 +662,60 @@ def _format_default_stimulus(self):
-                dividend_unit = self._get_desired_units(d_eqs[i]['lhs'].args[0])
-                divisor_unit = self._get_desired_units(d_eqs[i]['lhs'].args[1][0])
-                units = dividend_unit / divisor_unit
-                units_from = self._model.units.summarise_units(d_eqs[i]['lhs'])
-                if units.dimensionality == units_from.dimensionality:
-                    factor = self._model.units.get_conversion_factor(units, from_unit=units_from)
-                    d_eqs[i]['factor'] = factor
-
-        for i in range(len(d_eqs)):
-            if d_eqs[i]['factor'] != 1.0:
-                for j in range(i + 1, len(d_eqs)):
-                    d_eqs[j]['rhs'] =\
-                        d_eqs[j]['rhs'].subs({d_eqs[i]['lhs']:
-                                              (1 / d_eqs[i]['factor'] * d_eqs[i]['lhs'])})
-        for i in range(len(d_eqs)):
-            if d_eqs[i]['factor'] != 1.0:
-                d_eqs[i]['rhs'] = sp.sympify(d_eqs[i]['factor'] * d_eqs[i]['rhs'], evaluate=False)
-            d_eqs[i]['rhs'] = self._printer.doprint(d_eqs[i]['rhs'])
-            d_eqs[i]['lhs'] = self._printer.doprint(d_eqs[i]['lhs'])
-
-        return d_eqs
-
-    def _format_ode_system_info(self):
-        """ Format general ode system info for chaste output"""
+        return [{'lhs': self._printer.doprint(eqs.lhs),
+                 'rhs': self._printer.doprint(eqs.rhs),
+                 'units': self._model.units.summarise_units(eqs.lhs),
+                 'in_membrane_voltage': eqs not in self._derivative_eqs_exlc_voltage,
+                 'is_voltage': isinstance(eqs.lhs, sp.Derivative) and eqs.lhs.args[0] == self._membrane_voltage_var}
+                for eqs in self._derivative_equations]
+
+    def _format_free_variable(self):
+        """ Format free variable for chaste output"""
         ontology_terms = self._model.get_ontology_terms_by_symbol(self._time_variable, namespace_uri=self._OXMETA)
         name = ontology_terms[0] if len(ontology_terms) > 0 else self._name_printer.doprint(self._time_variable)
-        free_variable = {'name': name,
-                         'units': self._get_desired_units(self._time_variable),
-                         'system_name': self._model.name,
-                         'var_name': self._printer.doprint(self._time_variable)}
-        system_info = \
-            [{'name': self._model.get_ontology_terms_by_symbol(var, self._OXMETA)[0]
-              if self._model.has_ontology_annotation(var, self._OXMETA) else self._name_printer.doprint(var),
-              'initial_value': str(self._model.get_initial_value(var))
-                if var not in self._state_var_conversion_factors
-                else str(self._state_var_conversion_factors[var] * self._model.get_initial_value(var)),
-              'units': self._get_desired_units(var)}
-             for var in self._state_vars]
+        return {'name': name,
```

Comment:
Could you just use `var_display_name` to calculate `name` rather than repeating checking ontology logic etc? (Perhaps not in this PR!)

Reply:
good spot!